clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBStoragePartition.java

```diff
@@ -548,6 +552,79 @@ public byte[] get(ByteBuffer keyBuffer) {
     }
   }
 
+  public List<byte[]> multiGet(List<byte[]> keys) {
+    readCloseRWLock.readLock().lock();
+    try {
+      makeSureRocksDBIsStillOpen();
+      return rocksDB.multiGetAsList(keys);
+    } catch (RocksDBException e) {
+      throw new VeniceException("Failed to get value from store: " + storeName + ", partition id: " + partitionId, e);
+    } finally {
+      readCloseRWLock.readLock().unlock();
+    }
+  }
+
+  public List<ByteBuffer> multiGet(List<ByteBuffer> keys, List<ByteBuffer> values) {
+    readCloseRWLock.readLock().lock();
+    try {
+      makeSureRocksDBIsStillOpen();
+      List<ByteBufferGetStatus> statusList = rocksDB.multiGetByteBuffers(keys, values);
+
+      List<ByteBuffer> resultList = new ArrayList<>(keys.size());
+      Iterator<ByteBufferGetStatus> statusIter = statusList.iterator();
+      Iterator<ByteBuffer> keyIter = keys.iterator();
+      ListIterator<ByteBuffer> valueIter = values.listIterator();
+      while (keyIter.hasNext()) {
+        if (!statusIter.hasNext() || !valueIter.hasNext()) {
```

Comment:
This check can be done once outside of the loop.

Reply:
Sure